### PR TITLE
bgpsyncer: only watch affinity blocks for a specific node

### PR DESCRIPTION
**Pull request by neiljerram**

### CODE REVIEW

File: lib/backend/syncersv1/bgpsyncer/bgpsyncer.go [hunk 1]
```diff
@@ -57,9 +57,11 @@ func New(client api.Client, callbacks api.SyncerCallbacks, node string, watchAll
 			ListInterface:   model.ResourceListOptions{Kind: apiv3.KindBGPPeer},
 			UpdateProcessor: updateprocessors.NewBGPPeerUpdateProcessor(),
 		},
-		{
+	}
+	if node != "" {
```
Comment (robbrockbank) [inline lib/backend/syncersv1/bgpsyncer/bgpsyncer.go]: Maybe worth adding a comment here that we only expect node to be blank for the RR and in that case we don't need to monitor the BlockAffinities

**Comment by neiljerram** [comment]
Thanks @robbrockbank I added a comment.  Have I got the details there right?

File: lib/backend/syncersv1/bgpsyncer/bgpsyncer.go [hunk 2]
```diff
@@ -57,9 +57,16 @@ func New(client api.Client, callbacks api.SyncerCallbacks, node string, watchAll
 			ListInterface:   model.ResourceListOptions{Kind: apiv3.KindBGPPeer},
 			UpdateProcessor: updateprocessors.NewBGPPeerUpdateProcessor(),
 		},
-		{
+	}
+	// When this syncer is used (via confd) in calico/node, it needs to know the affinity blocks
```
Comment (robbrockbank) [inline lib/backend/syncersv1/bgpsyncer/bgpsyncer.go]: the affinity blocks are actually just used to set up the blackhole routes - nothing to do with IPIP.

Comment (ozdanborne) [reply lib/backend/syncersv1/bgpsyncer/bgpsyncer.go]: addressed

**Pull request merged**